servers/src/grin/server.rs

```diff
@@ -451,18 +451,10 @@ impl Server {
 
 					height += 1;
 
-					// We need to query again for the actual block hash, as
-					// the difficulty iterator doesn't contain enough info to
-					// create a hash
-					// The diff iterator returns 59 block headers 'before' 0 to give callers
-					// enough detail to calculate initial block difficulties. Ignore these.
-					let block_hash = if height < 0 {
-						ZERO_HASH
+					let block_hash = if let Some(nh) = next.hash {
```

Comment:
syntactic sugar nitpicking: I think we could get rid of this branching by doing something like `block_hash: next.hash.unwrap_or(ZERO_HASH)` (might be a bit different in rust)

Reply:
:+1: